frontend/src/contexts/ApiProvider.tsx

```diff
@@ -46,6 +46,13 @@ const makeApiContext = () => {
   const api = createMemo(() => new Api(apiConfig()))
   createEffect(() => {
     writeApiDetails(apiConfig())
+
+    api().getServerInfo()
```

Comment:
This should not be needed as info is gathered on initial request to api and stored for the current session. It may however be `undefined` whilst it is being gathered or no valid api server is selected.

Reply:
Unfortunately, that is not the case. `getServerInfo()` never gets called on the home page, for unauthenticated session, at least.
That's what had me confused at first. Can be seen pretty easily just by opening the dev tools before opening the page. :sweat_smile:

Reply:
Thanks for finding a bug! Instead of your fix, could you implement the below in `frontend/src/index.tsx`:

```diff
 render(() => (
   <Router>
     <Route path="/" component={Wrapper}>
-      <Route path="/" component={Home} />
+      <Route path="/" component={() => <RequireApiSetupGuard><Home /></RequireApiSetupGuard>} />
       <Route path="/scratch-pad" component={ScratchPad} />
       <Route component={RequireNoAuthGuard}>
         <Route path="/signup" component={() => <RequireSignupAllowedGuard><Signup /></RequireSignupAllowedGuard>} />
```

This should ensure info is requested correctly.

Reply:
Sure! Thanks for looking into it, looks like a lot cleaner solution!

Reply:
Just tried it out - a side-effect of this is that the user now get's directly redirected to the login page if the server is unreachable, due to [this line](https://github.com/enchant97/note-mark/blob/8d499a791653da7d2de41ece9d90b2cc6e94cc1f/frontend/src/contexts/ApiProvider.tsx#L104).

Thinking about it, it makes sense tho - if the API is unreachable, the only really useful thing to do is setting the API URL anyway, so redirecting to the login page isn't wrong IMO.